src/system/mod.rs

```diff
@@ -0,0 +1,364 @@
+        Self::new_with_connector_zipped(
+            connector,
+            pkeys.into_iter().zip(candidates),
+        )
+        .await
+    }
+
+    /// Add a new peer into the `System` using the provided `Candidate` and
+    /// `Connector`
+    pub async fn add_peer<CD, C>(
+        &mut self,
+        connector: &C,
+        candidates: &[CD],
+        public: &PublicKey,
+    ) -> Result<(), ConnectError>
+    where
+        CD: fmt::Display + Send + Sync,
+        C: Connector<Candidate = CD>,
+    {
+        let connection = connector.connect_any(public, candidates).await?;
+
+        self.connections.insert(*public, connection);
+
+        Ok(())
+    }
+
+    /// Add many peers to this `System` using the provided `Connector`
+    pub async fn add_peers<CD, C>(
+        &mut self,
+        connector: &C,
+        candidates: &[(CD, PublicKey)],
+    ) -> impl Iterator<Item = (PublicKey, ConnectError)>
+    where
+        CD: fmt::Display + Send + Sync,
+        C: Connector<Candidate = CD>,
+    {
+        let (ok, err): (Vec<_>, Vec<_>) = connector
+            .connect_many(candidates)
+            .await
+            .drain(..)
+            .zip(candidates.iter().map(|x| x.1))
+            .map(|(result, pkey)| match result {
+                Ok(connection) => {
+                    info!("connected to {}", pkey);
+                    Ok((pkey, connection))
+                }
+                Err(e) => {
+                    error!("failed to connect to {}: {}", pkey, e);
+                    Err((pkey, e))
+                }
+            })
+            .partition(Result::is_ok);
+
+        self.connections.extend(ok.into_iter().map(Result::unwrap));
+
+        err.into_iter().map(Result::unwrap_err)
+    }
+
+    /// Add a `Listener` to this `System` that will accept incoming peer
+    /// `Connection`s
+    pub async fn add_listener<C, L>(
+        &mut self,
+        mut listener: L,
+    ) -> impl Stream<Item = ListenerError>
+    where
+        C: fmt::Display + Sync + Send,
+        L: Listener<Candidate = C> + 'static,
+    {
+        let (err_tx, err_rx) = mpsc::channel(1);
+        let (peer_tx, peer_rx) = mpsc::channel(32);
+
+        let handle =
+            task::spawn(async move {
+                loop {
+                    match listener.accept().await {
+                        Err(e) => {
+                            if let Err(e) = err_tx.send(e).await {
+                                warn!(
+                                    "lost error from listener on {}: {}",
+                                    listener.local_addr().unwrap_or_else(
+                                        || (Ipv4Addr::UNSPECIFIED, 0).into()
+                                    ),
+                                    e,
+                                );
+                            }
+                        }
+                        Ok(connection) => {
+                            let _ = peer_tx.send(connection).await;
```

Comment:
no error handling?

Reply:
I found some more `let _`: `rg 'let _'`
```
tests/net/mod.rs
23:        let _ = tracing::subscriber::set_global_default(subscriber);

src/system/mod.rs
232:                            let _ = peer_tx.send(connection).await;
341:        let _ = system

src/net/listener/directory.rs
153:        let _ = self.exit_tx.send(());

src/test/log.rs
12:        let _ = subscriber::set_global_default(subscriber);
```
please add a comment on each as to why you're dropping the error (or don't drop it, that's even better)